### PR TITLE
DOC: Deployment template doc. update

### PR DESCRIPTION
**Pull request by Shubham9t9**
# Description

Documentation for deployment template

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# Checklist:

* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -20,8 +20,8 @@
 * [Creating Application](user-guide/creating-application/README.md)
   * [Git Repository](user-guide/creating-application/git-material.md)
   * [Docker Build Configuration](user-guide/creating-application/docker-build-configuration.md)
-  * [Deployment Template](user-guide/creating-application/deployment-template.md)
-    * [Rollout](user-guide/creating-application/deployment-template.md)
+  * [Deployment Template](user-guide/creating-application/deployment-template/README.md)
```
Comment (prakarsh-dt) [inline docs/SUMMARY.md:23]: DO NOT CHANGE this link as it is hardcoded inside Product!

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -20,8 +20,8 @@
...
+    * [Rollout](user-guide/creating-application/deployment-template/deployment-template.md)
```
Comment (prakarsh-dt) [inline docs/SUMMARY.md:24]: It should be
[Rollout Deployment](user-guide/creating-application/deployment-template/rollout-deployment.md)

Comment (Shubham9t9) [reply docs/SUMMARY.md]: changed

File: docs/SUMMARY.md (hunk 1 above)
Comment (Shubham9t9) [reply docs/SUMMARY.md]: Changed to as it was

File: docs/user-guide/creating-application/deployment-template.md [hunk 3]
```diff
@@ -1,785 +1,7 @@
-# Devtron Charts
+# Deployment Template
```
Comment (prakarsh-dt) [inline docs/user-guide/creating-application/deployment-template.md:1]: This will be 
Devtron Charts

Comment (Shubham9t9) [reply docs/user-guide/creating-application/deployment-template.md]: changed

**Review by prakarsh-dt — APPROVED** [review]

**Pull request merged**